kubespawner/spawner.py

```diff
@@ -781,6 +783,46 @@ def _hub_connect_port_default(self):
         """
     )
 
+    form_template = Unicode(
+        """<label for="profile">Please select a profile for your Jupyter:</label>
+        <select class="form-control" name="profile" required autofocus>
+        {input_template}
+        </select>
+        """,
+        config = True,
+        help = """Template to use to construct options_form text. {input_template} is replaced with
+            the result of formatting input_template against each item in the profiles list."""
+        )
+
+    input_template = Unicode("""
+        <option value="{key}" {first}>{display}</option>""",
+        config = True,
+        help = """Template to construct {input_template} in form_template. This text will be formatted
+            against each item in the profiles list, in order, using the following key names:
+            ( display, key, type ) for the first three items in the tuple, and additionally
+            first = "checked" (taken from first_template) for the first item in the list, so that
+            the first item starts selected."""
+        )
+
+    first_template = Unicode('selected',
+        config=True,
+        help="Text to substitute as {first} in input_template"
+        )
+
+    options_form = Unicode()
+    
+    single_user_profile_list = List(
```

Comment:
Can this be `profile_list`? I want to move toward removing the `singleuser_` prefix from config, since everything here is necessarily about configuring singleuser pods, it's redundant.

Reply:
done